tests/test_flags/test_dry_run.py

```
@@ -0,0 +1,524 @@
+
+                                # Verify auto-stage was passed through
+                                mock_show_summary.assert_called_once()
+                                args = mock_show_summary.call_args[0][0]
+                                assert args.all is True
+
+    def test_dry_run_combined_with_verbose(self):
+        """Test that --dry-run and -v can be used together."""
+        test_argv = ["git-commitai", "--dry-run", "-v"]
+
+        with patch("sys.argv", test_argv):
+            with patch("subprocess.run") as mock_subprocess:
```

Ruff (0.12.2)

Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)